lib/grandpa/grandpa.go

```diff
@@ -446,191 +403,112 @@ func (s *Service) primaryBroadcastCommitMessage() {
 	s.network.GossipMessage(msg)
 }
 
-// playGrandpaRound executes a round of GRANDPA
-// at the end of this round, a block will be finalised.
-func (s *Service) playGrandpaRound() error {
-	logger.Debugf("starting round %d with set id %d",
-		s.state.round, s.state.setID)
-	start := time.Now()
-
-	ctx, cancel := context.WithCancel(s.ctx)
-	defer cancel()
-
-	isPrimary, err := s.handleIsPrimary()
+func (s *Service) checkRoundCompletable() (bool, error) {
```

Comment:
In future this will be extended to account for potential votes, yes?

Reply:
Right!